### DISCUSSION TITLE
`onClick` on `Link` does not get called in test

### DISCUSSION THREAD

**Discussion by donovanlopez** [vercel/next.js#73465] [Help]
### Summary

I'm using v12 with the `experimental.newNextLinkBehavior` flag on. I'm trying to write a test with to see if the `onClick` passed into my wrapper link component is called but the `mockOnClick` doesn't seem to get called. It gets called when I actually use the component and click on it but in the test is doesn't get called. Is there something I'm missing behind how `onClick` works with the `Link` component?

```
  it('calls mockOnClick', async () => {
    const mockOnClick = jest.fn();

    mockRouter.setCurrentUrl('/path/1');

    const { user } = render(
      <WrappedLink href='/path/2' onClick={mockOnClick}>
        Test link
      </WrappedLink>,
      { wrapper: MemoryRouterProvider },
    );

    await user.click(screen.getByText('Test link'));
    
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

```

**Comment by icyJoseph** [comment]
How does WrappedLink look like? IIRC, it has to forwardRefs, or something like that? And take some of the props passed onto it?

Also, in the test environment, you probably need to tell Next.js, that you are using the experimental feature. I think you need to set this `__NEXT_NEW_LINK_BEHAVIOR` env var to true, or w.e. in your test env, as well.

**Reply by donovanlopez** [reply]
This is what the WrappedLink looks like basically. There's some logic that I left out but this is basically how the `Link` is used. I'm not using any forwardRefs in it.

```
export const WrappedLink = ({ children, href, onClick }: WrappedLinkProps) => {
  return (
    <Link href={href} passHref onClick={onClick}>
      {children}
    </Link>
  );
};``

Ah ok I think I might've missed setting `__NEXT_NEW_LINK_BEHAVIOR` but I tried doing `process.env.__NEXT_NEW_LINK_BEHAVIOR = 'true'` in my `jest.setup.tsx` but it can't find `__NEXT_NEW_LINK_BEHAVIOR`.
